core/src/main/scala/com.softwaremill.kmq/redelivery/CommitMarkerOffsetsActor.scala

```diff
@@ -1,56 +1,75 @@
 package com.softwaremill.kmq.redelivery
 
-import akka.actor.Actor
-import com.softwaremill.kmq.KafkaClients
-import com.typesafe.scalalogging.StrictLogging
-import org.apache.kafka.clients.consumer.OffsetAndMetadata
+import cats.effect.{IO, Resource}
+import com.softwaremill.kmq.KafkaClientsResourceHelpers
+import org.apache.kafka.clients.consumer.{KafkaConsumer, OffsetAndMetadata}
 import org.apache.kafka.common.TopicPartition
 import org.apache.kafka.common.serialization.ByteArrayDeserializer
 
-import scala.collection.JavaConverters._
+import scala.jdk.CollectionConverters._
 import scala.concurrent.duration._
+import fs2.Stream
+import cats.effect.std.Queue
 
-class CommitMarkerOffsetsActor(markerTopic: String, markerOffsetGroupId: String, clients: KafkaClients) extends Actor with StrictLogging {
+import org.typelevel.log4cats.slf4j.Slf4jLogger
 
-  private val consumer = clients.createConsumer(markerOffsetGroupId, classOf[ByteArrayDeserializer], classOf[ByteArrayDeserializer])
-
-  private var toCommit: Map[Partition, Offset] = Map()
-
-  import context.dispatcher
+class CommitMarkerOffsetsActor(
```

Comment:
[minor] maybe this should have a private constructor as the only way to create is via `create`

Reply:
fixed